readme.md

```diff
@@ -17,39 +17,81 @@ $ npm install alpha-sort
 ```js
 const alphaSort = require('alpha-sort');
 
-['b', 'a', 'c'].sort(alphaSort.ascending);
+['b', 'a', 'c'].sort(alphaSort());
 //=> ['a', 'b', 'c']
+
+['b', 'a', 'c'].sort(alphaSort({descending: true}));
+//=> ['c', 'b', 'a']
+
+['B', 'a', 'C'].sort(alphaSort({caseInsensitive: true}));
+//=> ['a', 'B', 'C']
+
+['file10.txt', 'file2.txt', 'file03.txt'].sort(alphaSort({natural: true}));
+//=> ['file2.txt', 'file03.txt', 'file10.txt']
 ```
 
 
 ## API
 
-### alphaSort.ascending
+### alphaSort(options?)
+
+Get a comparator function to be used as argument for [`Array#sort`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).
+
+#### options
 
-Ascending sort comparator.
+Type: `object`
 
-### alphaSort.descending
+##### descending
 
-Descending sort comparator.
+Type: `boolean`\
+Default: `false`
 
-### alphaSort.caseInsensitiveAscending
+Whether or not to sort descending.
 
-Case-insensitive ascending sort comparator.
+##### caseInsensitive
 
-Note: If two elements are considered equal in the case-insensitive comparison, the tie-break will be a case-sensitive comparison:
+Type: `boolean`\
+Default: `false`
+
+Whether or not to sort in a case-insensitive way.
+
+Note: If two elements are considered equal in the case-insensitive comparison, the tie-break will be a standard (case-sensitive) comparison. Example:
 
 ```js
 const alphaSort = require('alpha-sort');
 
-['bar', 'baz', 'Baz'].sort(alphaSort.caseInsensitiveAscending);
+['bar', 'baz', 'Baz'].sort(alphaSort({caseInsensitive: true}));
 //=> ['bar', 'Baz', 'baz']
 ```
 
-### alphaSort.caseInsensitiveDescending
+##### natural
+
+Type: `boolean`\
+Default: `false`
+
+Whether or not to sort using [natural sort order](https://en.wikipedia.org/wiki/Natural_sort_order).
 
-Case-insensitive descending sort comparator.
+Note: If two elements are considered equal in the natural sort order comparison, the tie-break will be a standard (non-natural) comparison.
+
+##### preprocessor
+
+Type: `function`\
+Default: `undefined`
+
+A custom function that you can provide to manipulate the elements before sorting. This does not modify the values of the array; it only interferes in the sorting order.
+
+This can be used, for example, if you are sorting book titles in English and want to ignore common articles such as `the`, `a` or `an`:
+
+```js
+const alphaSort = require('alpha-sort');
+
+['The Foo', 'Bar'].sort(alphaSort({
+	preprocessor: title => title.replace(/^(the|a|an) /i, '')
+}));
+//=> ['Bar', 'The Foo']
+```
 
-The same note for `caseInsensitiveAscending` applies.
+Note: If two elements are considered equal when sorting with a custom preprocessor, the tie-break will be a comparison without the custom preprocessor.
```

Comment:
What's the reasoning behind this? Wouldn't it be better to just do stable sort then and not change any of the positions?

Reply:
I wanted the sorted result to not depend on the original order, only the original values. Therefore, if someone shuffles the array before calling `.sort(alphaSort(...))`, the output will be the same as if they didn't.

This property was on my mind because recently I needed to test a function that returns an array but does not guarantee the return order. So I sorted the result before comparing it to an array I defined as `expected`. For this, I prefer a sort method with the above property rather than a stable sort.

Also, you wrote this comment for the `preprocessor` part, but please note that I guaranteed the above property for `caseInsensitive` and `natural` as well (and therefore they are also not stable right now). If you prefer a stable sort, would you want those sorts to be stable too? Or only the ones using a custom preprocessor?

Reply:
Hmm, difficult problem, because I do think it makes sense to fall back to normal sorting when using `natural` and `caseInsensitive`. So I guess this is fine here too as long as it's clearly documented, which it is.

Reply:
Awesome, I think a possible next step would be to add another `stable` option that controls if tie-breaks are done in the way of this PR or if they are done based on the indexes. I can create an issue to track this after this PR is merged.